lib/ansible/playbook/role_include.py

```diff
@@ -88,8 +88,9 @@ def get_block_list(self, play=None, variable_manager=None, loader=None):
         actual_role = Role.load(ri, myplay, parent_role=self._parent_role, from_files=from_files,
                                 from_include=True, validate=self.rolespec_validate, public=self.public)
         actual_role._metadata.allow_duplicates = self.allow_duplicates
+        actual_role.static = self.statically_loaded
 
-        if self.statically_loaded or self.public:
+        if self.public or (self.public is None and (C.DEFAULT_PRIVATE_ROLE_VARS is False or (C.DEFAULT_PRIVATE_ROLE_VARS is None and actual_role.static))):
```

Comment:
This should be rewritten or explained. 😅

Reply:
just did above, but yes, a comment is needed as it will be confusing for anyone looking at the code